lisa/wa.py

```diff
@@ -168,7 +168,7 @@ def __init__(self, path, kernel_path=None):
         auto_collectors = {
             name: cls
             for name, cls in collector_classes.items()
-            if not self._needs_params(cls)
+            if not self._needs_params(cls) or cls._FORCE_AUTO
```

Comment:
That's broken, if a class needs some parameters you simply cannot create the collector with this `__getitem__` API as it would force calling a function with too few parameters.

Instead simply fix `_needs_params()` so it ignores parameters with `.kind in (VAR_POSITIONAL, VAR_KEYWORD)` : https://docs.python.org/3/library/inspect.html#inspect.Parameter.kind